modules/core/arrow-validation/src/test/kotlin/arrow/validation/refinedTypes/numeric/GreaterEqualTest.kt

```diff
@@ -0,0 +1,55 @@
+package arrow.validation.refinedTypes.numeric
+
+import arrow.data.Nel
+import arrow.data.Validated
+import arrow.data.ValidatedPartialOf
+import arrow.instances.nonemptylist.semigroup.semigroup
+import arrow.instances.order
+import arrow.instances.validated.applicativeError.applicativeError
+import arrow.test.UnitSpec
+import arrow.typeclasses.ApplicativeError
+import arrow.typeclasses.Order
+import arrow.validation.RefinedPredicateException
+import arrow.validation.refinedTypes.numeric.validated.greaterEqual.greaterEqual
+import io.kotlintest.KTestJUnitRunner
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.filter
+import io.kotlintest.properties.forAll
+import org.junit.runner.RunWith
+
+@RunWith(KTestJUnitRunner::class)
+class GreaterEqualTest : UnitSpec() {
+  init {
+
+    val min = 100
+    val LESS = { max: Int ->
+      object : Less<ValidatedPartialOf<Nel<RefinedPredicateException>>, Int> {
+        override fun ORD(): Order<Int> = Int.order()
+
+        override fun max(): Int = max
+
+        override fun applicativeError(): ApplicativeError<ValidatedPartialOf<Nel<RefinedPredicateException>>,
+          Nel<RefinedPredicateException>> =
+          Validated.applicativeError(Nel.semigroup())
+
+        override fun invalidValueMsg(a: Int): String = Less.errorMsg(a, max)
+      }
+    }
+
+    "Can create GreaterEqual for every number greater or equal than min defined by instance" {
+      forAll(GreaterEqualGen(min)) { x: Int ->
+        x.greaterEqual(LESS(min)).isValid
```

Comment:
I would like to have a function with the same signature as `less(Ord<Int>, max: Int)` instead of passing a `Less` instance 🤷‍♂️